force-app/main/default/lwc/geGatewaySelectWidget/__tests__/geGatewaySelectWidget.test.js

```diff
@@ -0,0 +1,298 @@
+    () => {
+        return { default: jest.fn() };
+    },
+    { virtual: true }
+);
+
+jest.mock(
+    "@salesforce/apex/GE_GiftEntryController.getGatewaysFromElevate",
+    () => {
+        return { default: jest.fn() };
+    },
+    { virtual: true }
+);
+
+describe("c-ge-gateway-select-widget", () => {
+    beforeEach(() => {
+        getGatewaysFromElevate.mockResolvedValue(JSON.stringify(gatewaysFromElevate));
+    });
+
+    afterEach(() => {
+        GeGatewaySettings.clearDecryptedElevateSettings();
+        jest.clearAllMocks();
+        clearDOM();
+    });
+
+    describe("gateway assignment mode", () => {
+        beforeEach(() => {
+            getGatewayAssignmentSettings.mockResolvedValue(JSON.stringify({
+                defaultGatewayId: DEFAULT_GATEWAY_ID,
+                defaultTemplateId: DEFAULT_TEMPLATE_ID,
+                gatewayAssignmentEnabled: true,
+            }));
+        });
+
+        it("renders widget contents when not default template", async () => {
+            GeGatewaySettings.setElevateSettings(null, SELECTED_GATEWAY_ID);
+            const element = createGeGatewaySelectWidget(null);
+            document.body.appendChild(element);
+            await flushPromises();
+
+            expect(element).toBeTruthy();
+            expect(getDiv(element)).toBeTruthy();
+        });
+
+        it("does not render widget contents when default template", async () => {
+            GeGatewaySettings.setElevateSettings(null, DEFAULT_TEMPLATE_ID);
+            const element = createGeGatewaySelectWidget(null);
+            document.body.appendChild(element);
+            await flushPromises();
+
+            expect(element).toBeTruthy();
+            expect(getDiv(element)).toBeFalsy();
+        });
+
+        it("displays spinner if no gateways found", async () => {
+            GeGatewaySettings.setElevateSettings(null, null);
+            getGatewaysFromElevate.mockResolvedValueOnce(null);
+            const element = createGeGatewaySelectWidget(null);
+            document.body.appendChild(element);
+            await flushPromises();
+
+            expect(getSpinner(element)).toBeTruthy();
+        });
+
+        it("displays gateways and payment methods when expanded", async () => {
+            GeGatewaySettings.setElevateSettings(null, null);
+            const element = createGeGatewaySelectWidget(null);
+            document.body.appendChild(element);
+            await flushPromises();
+
+            const expandGatewayControl = getExpandGatewayControl(element);
+            expect(expandGatewayControl).toBeTruthy();
+            expandGatewayControl.click();
+            await flushPromises();
+
+            expect(getComboBox(element)).toBeTruthy();
+            expect(getACHCheckBox(element)).toBeTruthy();
+            expect(getCCCheckBox(element)).toBeTruthy();
+        })
+
+        it("does not display gateways and payment methods when collapsed", async () => {
```

Comment:
This and the test above could probably be the same test, just need to add the expectations after it's expanded 🤷

Reply:
I had initially combined the tests, but thought it might be more clear what was being tested if they were separate. I'll put them back together.